code/object_detector/communication/nxt_usb.py

```diff
@@ -62,18 +62,16 @@ def write_data(self, data: Result) -> None:
         should react upon by moving the turret
         :param data: a result data
         """
-        if not self.endpoint:
-            self.init_usb_communication()
-        self.endpoint.write(bytes([data.delta_x & 0xFF,
-                                   data.delta_y & 0xFF,
+        self.endpoint.write(bytes([int(data.location.x) & 0xFF,
+                                   int(data.location.y) & 0xFF,
                                    data.timestamp & 0xFF,
                                    (data.timestamp >> 8) & 0xFF]))
 
     def disconnect(self) -> None:
         """
         This broadcasts a "TURNOFF" signal, and sets the endpoint to None
         """
-        if self.endpoint:
+        if hasattr(self,'endpoint'):
```

Comment:
Tror også der skal være et tjek på om endpoint er None, skriver fra telefon, så har ikke lige koden, men tror der er en assert endpoint not none i constructoren og hvis den assertion fejler eksekvere den denne del og fejler så når den skriver disconnect request fordi endpoint er none

Reply:
Also, erindre den blev kaldt fra destructoren, så den skal være private, eller indholdet flyttes til destructoren